### PR TITLE
Mockers now supports mocking external functions

### PR DESCRIPTION
**Pull request by kriomant**

### CODE REVIEW

File: src/t_mockers.rs [hunk 1]
```diff
@@ -140,11 +140,19 @@ impl TestSuite for Mockers {
     }
 
     fn foreign() {
-        unimplemented!()
```
Comment (asomers) [inline src/t_mockers.rs:143]: mock_derive can also mock external Rust functions, not just C functions.  Can Mockers do that too?

Comment (kriomant) [reply src/t_mockers.rs:143]: Do you mean `extern "Rust" { … }`? It works.

Comment (asomers) [reply src/t_mockers.rs:143]: Could you please add that to the test, so it's functionally equivalent to `t_mock_derive::t::foreign`?

**Comment by asomers** [comment]
LGTM!  Now, can you just squash the 2nd and 3rd commits?  You can leave the 1st commit separate.

**Comment by asomers** [comment]
Thanks for your submission!  I'm going to merge it now.  However, I won't be able to update the website for another day or two.

**Pull request merged**